worlds/sc2wol/LogicMixin.py

```diff
@@ -145,4 +151,83 @@ def _sc2wol_final_mission_requirements(self, multiworld: MultiWorld, player: int
+                       or self._sc2_hots_has_brood_lord(multiworld, player) \
+                       or advanced and (self.has('Infestor', player) or self._sc2_hots_has_viper(multiworld, player))
+        if core_unit and support_unit:
+            return True
+        vespene_unit = self.has_any({'Ultralisk', 'Aberration'}, player) \
+                       or advanced and self._sc2_hots_has_viper(multiworld, player)
+        return vespene_unit and self.has_any({'Zergling', 'Swarm Queen'}, player)
+
+    def _sc2_hots_has_basic_comp(self, multiworld: MultiWorld, player: int) -> bool:
+        if get_option_value(multiworld, player, 'game_difficulty') < 3 \
+           or self._sc2_hots_has_basic_kerrigan(multiworld, player) \
+           or self._sc2_hots_has_two_kerrigan_actives(multiworld, player):
+            return self._sc2_hots_has_common_unit(multiworld, player)
+        else:
+            return self._sc2_hots_has_competent_comp(multiworld, player)
+
+    def _sc2_hots_can_spread_creep(self, multiworld: MultiWorld, player: int) -> bool:
+        return get_option_value(multiworld, player, 'required_tactics') > 0 or self.has('Swarm Queen', player)
+    
+    def _sc2_hots_has_competent_defense(self, multiworld: MultiWorld, player: int) -> bool:
+        return self._sc2_hots_has_common_unit(multiworld, player) and \
+            ((self.has('Swarm Host', player) or self._sc2_hots_has_brood_lord(multiworld, player) or self._sc2_hots_has_impaler_or_lurker(multiworld, player)) or \
+            (get_option_value(multiworld, player, 'required_tactics') > 0 and (self._sc2_hots_has_viper(multiworld, player) or self.has('Spine Crawler', player))))
+
+    def _sc2_hots_has_basic_kerrigan(self, multiworld: MultiWorld, player: int) -> bool:
+        # One active ability that can be used to defeat enemies directly on Standard
+        if get_option_value(multiworld, player, "required_tactics") == 0 and \
+            not self.has_any({"Kinetic Blast (Kerrigan Tier 1)", "Leaping Strike (Kerrigan Tier 1)",
+                              "Crushing Grip (Kerrigan Tier 2)", "Psionic Shift (Kerrigan Tier 2)",
+                              "Spawn Banelings (Kerrigan Tier 4)"}, player):
+            return False
+        # Two non-ultimate abilities
+        count = 0
+        for item in ("Kinetic Blast (Kerrigan Tier 1)", "Leaping Strike (Kerrigan Tier 1)", "Heroic Fortitude (Kerrigan Tier 1)",
+                     "Chain Reaction (Kerrigan Tier 2)", "Crushing Grip (Kerrigan Tier 2)", "Psionic Shift (Kerrigan Tier 2)",
+                     "Spawn Banelings (Kerrigan Tier 4)", "Infest Broodlings (Kerrigan Tier 6)", "Fury (Kerrigan Tier 6)"):
+            if self.has(item, player):
+                count += 1
+            if count >= 2:
+                return True
+        return False
+
+    def _sc2_hots_has_two_kerrigan_actives(self, multiworld: MultiWorld, player: int) -> bool:
+        count = 0
+        for i in range(7):
```

Comment:
I agree it would fit better, but I don't know what the API for that looks like, and I believe there's a planned PR for adding item groups, so I think it's better to wait on this?

Reply:
Don't we have Upgrades/etc

Reply:
I had a look, some of the Terran upgrades have item groups, which just look for item type. I'm adding Zerg item groups, but this function specifically needs _active_ abilities. I would have to manually make an item group for those, but that would be duplicate information to `kerrigan_actives`, so I don't think it's worth it just for this function.

Reply:
I'd think more about using a separate item group. Therefore we'll get Actives, Passives and Economy Upgrades